pkg/ipam/poolallocator/allocator.go

```diff
@@ -272,6 +301,33 @@ func (a *IPPoolAllocator) AllocateNext(state v1alpha2.IPAddressPhase, owner v1al
 	return ip, subnetSpec, err
 }
 
+// AllocateStatefulSet pre-allocates continuous range of IPs for StatefulSet.
+// This functionality is useful when StatefulSet does not have a dedicated IP Pool assigned.
+// It returns error if such range is not available. In this case IPs for the StatefulSet will
+// be allocated on the fly, and there is no guarantee for continuous IPs.
+func (a *IPPoolAllocator) AllocateStatefulSet(namespace, name string, size int) error {
```

Comment:
Where is it called? If it's just a preparation for another feature, could you move it and all the callees it introduces to the PR that implements the feature? Otherwise it's hard to say whether the methods can really fit that feature without any caller and may introduce unnecessary changes.
This PR could focus on StatefulSet Pod IP cleanup which is scoped in 1.5?

Reply:
This is for future feature, and not used in code as of today, but used in tests. I'd prefer to include this in the PR for future use rather than copying this functionality in testing.